signature/cose/envelope.go

```diff
@@ -569,3 +572,21 @@ func contains(s []interface{}, e interface{}) bool {
 	}
 	return false
 }
+
+// parseTime parses time values from cose.ProtectedHeader
+// in go-cose,
+//  1. CBOR times (tag 0 and 1) decode to time.Time.
+//  2. decode CBOR uint/int to Go int64
+//
+// For more details: https://github.com/fxamacker/cbor/blob/7704fa5efaf3ef4ac35aff38f50f6ff567793072/decode.go#L52
+// https://github.com/veraison/go-cose/blob/f72b6bc06b90d1e9c07ab11332df92c7838e3027/cbor.go#L43
+func parseTime(timeValue interface{}) (time.Time, error) {
+	switch t := timeValue.(type) {
+	case time.Time:
+		return t, nil
+	case int64:
+		return time.Unix(t, 0), nil
```

Comment:
Per spec, we don't support integers without a tag, do we?

Reply:
> Per spec, we don't support integers without a tag, do we?

User input a time.Time into signRequest -> Sign -> go-cose Marshalls/Encodes this time.Time into an integer
-> go-cose Unmarshalls/Decodes this integer to int64 -> Verify. Therefore, during Verify, we still need to check the int64 type.

Reply:
Code updated per discussion.